Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -207,19 +200,12 @@ IEnumerator IEnumerable.GetEnumerator()
 
 		public bool TryGetValue(string key, out object value)
 		{
-			return _innerDictionary.TryGetValue(key, out value)
-				|| (_mergedInstance != null && _mergedInstance.TryGetValue(key, out value))
-				|| (MergedDictionaries != null && TryGetMergedDictionaryValue(key, out value));
-		}
-
-		bool TryGetMergedDictionaryValue(string key, out object value)
-		{
-			foreach (var dictionary in MergedDictionaries.Reverse())
-				if (dictionary.TryGetValue(key, out value))
-					return true;
-
-			value = null;
-			return false;
+			object outValue;
+			var containsKey = _innerDictionary.TryGetValue(key, out outValue)
+				|| _mergedInstance != null && _mergedInstance.TryGetValue(key, out outValue)
+				|| _mergedDictionaries != null && _mergedDictionaries.Reverse().Any(d => d.TryGetValue(key, out outValue));
+			value = outValue;
+			return containsKey;
```

Comment:
and here, no real point in merging

Reply:
Please, look at the property MergedResources  which uses Linq for similar _mergedDictionaries enumerating.
```javascript
		internal IEnumerable<KeyValuePair<string, object>> MergedResources {
			get {
				if (_mergedDictionaries != null)
					foreach (var r in _mergedDictionaries.Reverse().SelectMany(x => x.MergedResources))
						yield return r;
				if (_mergedInstance != null)
					foreach (var r in _mergedInstance.MergedResources)
						yield return r;
				foreach (var r in _innerDictionary)
					yield return r;
			}
		}
```
Now look at current implementation from master brunch of TryGetValue which uses custom method for enumeration.
```javascript
		public bool TryGetValue(string key, out object value)
		{
			return _innerDictionary.TryGetValue(key, out value)
			       || (_mergedInstance != null && _mergedInstance.TryGetValue(key, out value))
			       || (_mergedDictionaries != null && TryGetMergedDictionaryValue(key, out value));
		}

		bool TryGetMergedDictionaryValue(string key, out object value)
		{
			foreach (var dictionary in _mergedDictionaries.Reverse())
				if (dictionary.TryGetValue(key, out value))
					return true;

			value = null;
			return false;
		}
```

And compare with laconic Linq approach
```javascript
		public bool TryGetValue(string key, out object value)
		{
			var containsKey = _innerDictionary.TryGetValue(key, out var outValue)
				|| _mergedInstance != null && _mergedInstance.TryGetValue(key, out outValue)
				|| _mergedDictionaries != null && _mergedDictionaries.Reverse().Any(d => d.TryGetValue(key, out outValue));
			value = outValue;
			return containsKey;
		}
```

Seems, that second variant of TryGetValue looks more pretty..